src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -47,14 +49,19 @@ internal sealed class SerializableSourceText
     /// The hash that would be produced by calling <inheritdoc cref="SourceText.GetContentHash"/> on <see
     /// cref="_text"/>.  Can be passed in when already known to avoid unnecessary computation costs.
     /// </summary>
-    public readonly ImmutableArray<byte> ContentHash;
+    private readonly ImmutableArray<byte> _contentHash;
 
     /// <summary>
     /// Weak reference to a SourceText computed from <see cref="_storage"/>.  Useful so that if multiple requests
     /// come in for the source text, the same one can be returned as long as something is holding it alive.
     /// </summary>
     private readonly WeakReference<SourceText?> _computedText = new(target: null);
 
+    /// <summary>
+    /// Checksum of the contents (see <see cref="SourceText.GetContentHash"/>) of the text.
+    /// </summary>
+    public Checksum ContentChecksum => Checksum.Create(_contentHash);
```

Comment:
>Checksum.Create(_contentHash); [](http://example.com/codeflow?start=39&length=30)

Any reason why not to just store this as a Checksum instead of an ImmutableArray?

Reply:
done.